Adjust fetcher to API changes (#51)
neptune-ai/neptune-fetcher: README.md

| `progress_bar`    | `bool`, `Type[ProgressBarCallback]`, optional | `None`              | Set to `False `to disable the download progress bar, or pass a type of ProgressBarCallback to [use your own progress bar](https://docs.neptune.ai/usage/querying_metadata/#using-a-custom-progress-bar). If set to `None` or `True`, the default tqdm-based progress bar will be used.                                 |
| `query`           | `str`, optional                               | `None`              | NQL query string. Example: `"(accuracy: float > 0.88) AND (loss: float < 0.2)"`. Exclusive with the `with_ids`, `custom_ids`, `states`, `owners`, and `tags` parameters. For syntax, see [Neptune Query Language](https://docs.neptune.ai/usage/nql/) in Neptune docs.                                                 |                                                                                                                  |

__Returns:__ `pandas.DataFrame`: A pandas DataFrame containing metadata of the fetched runs.

> [!IMPORTANT]
> When using a regular expression to filter columns, the total number of matched fields must not exceed 5000.

> > For runs this limit is 100 matched entries.

Reviewer: Could you point me to where this is enforced? 
Btw, not sure if related, but from what I see, `MAX_REGEXABLE_RUNS` const is not used.

Also, this sentence was unclear to me - I read it as "when using a regex to filter columns in runs, the limit of columns is 100"

__Parameters:__

| Name              | Type                                          | Default             | Description                                                                                                                                                                                                                                                                                                            |
|-------------------|-----------------------------------------------|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `columns`         | `List[str]`, optional                         | `None`              | Names of columns to include in the table, as a list of field names. The Neptune Custom ID (`sys/custom_run_id`) is always included. If `None`, only Custom ID is included. **Note:** When using one or both of the `columns` and `columns_regex` parameters, the total number of matched columns must not exceed 5000. |

Reviewer 2: ```suggestion
| `columns`         | `List[str]`, optional                         | `None`              | Names of columns to include in the table, as a list of field names. The custom run identifier (`sys/custom_run_id`) is always included. If `None`, only the custom ID is included. **Note:** When using one or both of the `columns` and `columns_regex` parameters, the total number of matched columns must not exceed 5000. |
```

| `progress_bar`    | `bool`, `Type[ProgressBarCallback]`, optional | `None`              | Set to `False `to disable the download progress bar, or pass a type of ProgressBarCallback to [use your own progress bar](https://docs.neptune.ai/usage/querying_metadata/#using-a-custom-progress-bar). If set to `None` or `True`, the default tqdm-based progress bar will be used.                                 |
| `query`           | `str`, optional                               | `None`              | NQL query string. Example: `"(accuracy: float > 0.88) AND (loss: float < 0.2)"`. Exclusive with the `with_ids`, `custom_ids`, `states`, `owners`, and `tags` parameters. For syntax, see [Neptune Query Language](https://docs.neptune.ai/usage/nql/) in Neptune docs.                                                 |                                                                                                                  |

__Returns:__ `pandas.DataFrame`: A pandas DataFrame containing metadata of the fetched runs.

> [!IMPORTANT]
> When using a regular expression to filter columns, the total number of matched fields must not exceed 5000.

> > Column `sys/custom_run_id` and what is passed as `sort_by` are always included.

Reviewer 2: ```suggestion
> The following fields are always included:
>
> - `sys/custom_run_id`: the custom run identifier.
> - The field to sort by. That is, the field name passed to the `sort_by` argument.
```

__Parameters__:

| Name              | Type                                          | Default             | Description                                                                                                                                                                                                                                                                                                                                                        |
|-------------------|-----------------------------------------------|---------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `columns`         | `List[str]`, optional                         | `None`              | Names of columns to include in the table, as a list of field names. The Neptune Custom ID (`sys/custom_run_id`) and Experiment Name (`sys/name`) are always included. If `None`, only Custom ID and Name are included. **Note:** When using one or both of the `columns` and `columns_regex` parameters, the total number of matched columns must not exceed 5000. |

Reviewer 2: ```suggestion
| `columns`         | `List[str]`, optional                         | `None`              | Names of columns to include in the table, as a list of field names. The custom run identifier (`sys/custom_run_id`) and experiment name (`sys/name`) are always included. If `None`, only the custom ID and name are included. **Note:** When using one or both of the `columns` and `columns_regex` parameters, the total number of matched columns must not exceed 5000. |
```
